tests/getopt/main.c

```diff
@@ -3,6 +3,14 @@
 
 #include "getopt.h"
 
+/*
+ * The test suite ignores the exit code of this program, so that we can
```

Comment:
Umm, what?  Looks to me like we're checking exit codes...?

Reply:
Kind-of.  If there's an exit code, the test shell script has an early exit, but it's called via:

    @/bin/sh -e test_getopt.sh 2>test_getopt.log || true

https://github.com/Tarsnap/libcperciva/blob/master/tests/getopt/Makefile#L7

The thing that ultimately produces a "FAILED!" from `make test` is:

    @if cmp -s test_getopt.log test_getopt.good; then	\

... but I just noticed that this doesn't actually generate a non-zero exit code.  That seems like it'd be a good idea, particularly with Travis-CI running "make test".

I'll clarify the comment, and get some `exit 1` in there.

Reply:
Just as a sanity check that I wasn't crazy: I changed the final line of `int main()` in `getopt/main.c` to `return (1);`, and still got:

```
$ make test
./test_getopt.sh
Testing command-line options parsing... PASSED!
```

because `test_getopt.log` still matched `test_getopt.good`.

Reply:
Never mind, I see that I misunderstood the effect of `|| true` when running the script within a Makefile, vs. calling a function within a script.